lib/github.py

```diff
@@ -0,0 +1,71 @@
+from __future__ import annotations
+
+import os
+import tempfile
+
+from devenv.constants import CI
+from devenv.constants import home
+from devenv.lib import fs
+from devenv.lib import proc
+
+
+def check_ssh_access(git: str) -> bool:
+    if CI:
+        return True
+    try:
+        # The remote returns code 1 when successfully authenticated.
+        proc.run(("ssh", "-T", "git@github.com"))
```

Comment:
Wouldn't it be cleaner to get the result and check the code for `1` instead of catching an error and checking for a string in the output?

Reply:
Is it worth returning a message along with False? It seems like the two tests are checking different things and the context is helpful to the user (either the SSH key isn't added to their profile correctly or they haven't setup SSO correctly.

Reply:
no, there are probably other scenarios where github returns 1 and has a different error message - we're specifically interested in this one

Reply:
added a print in the case that ssh key isn't added correctly
and SSO not working implies the ssh key is setup correctly